Assets/MixedRealityToolkit.SDK/Experimental/HandCoach/Scripts/InteractionHint.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections;
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    public class InteractionHint : MonoBehaviour
+    {
+        private GameObject VisualsRoot;
+
+        [Tooltip("React to hand tracking state to hide visuals when hands are being tracked.  If false, only the customShouldHideHands function will be evaluated.")]
+        public bool checkHandTracking = true;
+
+        [Tooltip("Min/max delay for showing the visuals.  If the user's hands are not in view, the visuals will appear after min seconds." +
+            "  If the user's hands are in view, the min timer will reset to 0, but the visuals will appear after max seconds")]
+        public Vector2 MinMaxDelay = new Vector2(5f, 10f);
+
+        [Tooltip("Uncheck this if you don't want to use max timer, only show hint when hands are out of view")]
+        public bool useMaxTimer = true;
+
+        [Tooltip("Number of times to repeat the hint before fading out")]
+        public int repeats = 2;
+
+        [Tooltip("If true, logic runs whenever this component is active.  If false, you must manually start the logic with StartShowTimer.")]
+        public bool autoActivate = true;
+
+        [Tooltip("Name of animation to play during loop")]
+        public string animationState;
+
+        [Tooltip("Time to wait between repeats")]
+        [SerializeField]
+        public float repeatDelay = 1f;
+
+        private string fadeInAnimationState = "Fade_In";
+
+        private string fadeOutAnimationState = "Fade_Out";
+
+        private float animationHideTime = 3f;
+
+        private float animationHideDuration = 0.5f;
+
+        private IMixedRealityHandJointService HandJointService => handJointService ?? (handJointService = MixedRealityToolkit.Instance.GetService<IMixedRealityHandJointService>());
+
+        private IMixedRealityHandJointService handJointService = null;
+
+        // Custom function to determine visibility of visuals.  Return true to hide visuals and reset min timer (max timer will still be in effect), return false when user is doing nothing and needs a hint.
+        public Func<bool> customShouldHideVisuals = delegate { return false; };
+
+        private Animator m_animator;
+
+        private bool m_animatingOut = false;
```

Comment:
For consistency, please don't use hungarian notation, as the rest of the mrtk code does not use it. Additionally, many of your other private fields don't use hungariant notation.